src/s3_fetchers.rs

```diff
@@ -1,84 +1,102 @@
 /// that represent a block heights
 pub(crate) async fn list_blocks(
-    client: &Client,
-    bucket: &str,
-    start_after_block_height: Option<String>,
-    continuation_token: Option<String>,
-) -> Result<crate::types::ListObjectResponse, Error> {
-    let response = client
+    s3_client: &Client,
+    s3_bucket_name: &str,
+    start_from_block_height: String,
+) -> anyhow::Result<Vec<String>> {
+    let response = s3_client
         .list_objects_v2()
-        .max_keys(10)
+        .max_keys(1000)
         .delimiter("/".to_string())
-        .set_start_after(start_after_block_height)
-        .set_continuation_token(continuation_token)
-        .bucket(bucket)
+        .start_after(start_from_block_height)
+        .bucket(s3_bucket_name)
         .send()
         .await?;
 
-    let continuation_token = response.next_continuation_token().map(ToOwned::to_owned);
-    let folder_names = match response.common_prefixes() {
+    Ok(match response.common_prefixes() {
         None => vec![],
         Some(common_prefixes) => common_prefixes
-            .into_iter()
+            .iter()
             .filter_map(|common_prefix| common_prefix.prefix.clone())
             .collect(),
-    };
-
-    Ok(crate::types::ListObjectResponse {
-        continuation_token,
-        folder_names,
     })
 }
 
-/// By the given block height (`key`) gets the objects:
+/// By the given block height (`block_height_prefix`) gets the objects:
 /// - block.json
 /// - shard_N.json
 /// Reads the content of the objects and parses it to JSON.
 /// Returns the result in a single JSON
-pub(crate) async fn get_object(
-    client: &Client,
-    bucket: &str,
-    key: &str,
-    tracked_shards: &Vec<u8>,
-) -> Result<serde_json::Value, Error> {
-    let mut main_json = serde_json::json!({});
-    let block_json = {
-        let response = client
-            .get_object()
-            .bucket(bucket)
-            .key(format!("{}block.json", key))
-            .send()
-            .await?;
+pub(crate) async fn fetch_streamer_message(
+    s3_client: &Client,
+    s3_bucket_name: &str,
+    block_height_prefix: &str,
+) -> anyhow::Result<near_indexer_primitives::StreamerMessage> {
+    let block_json: serde_json::Value = {
+        let response = loop {
+            if let Ok(response) = s3_client
+                .get_object()
+                .bucket(s3_bucket_name)
+                .key(format!("{}block.json", block_height_prefix))
+                .send()
+                .await
+            {
+                break response;
+            }
+        };
 
         let body_bytes = response.body.collect().await.unwrap().into_bytes();
 
         serde_json::from_slice(body_bytes.as_ref()).unwrap()
     };
 
-    main_json["block"] = block_json;
+    let shards_num: u64 = block_json["header"]["chunks_included"].as_u64().unwrap();
+
+    let mut main_json = serde_json::json!({
+        "block": block_json,
+    });
```

Comment:
I would just deserialize it straight to `IndexerBlock` type here instead of dealing with JSON

Reply:
We don't have `IndexerBlock` we're using `nearcore`s `BlockView` there, I'm not sure we want to introduce a new struct at the moment, no?

Reply:
I meant whatever type is there `BlockView` is fine, I thought we had a custom one there. It does not matter. My point was to deserialize it immediately to the proper type instead of going through `as_u64().unwrap()` and then crafting `main_json`